airflow/models.py

```diff
@@ -3367,6 +3367,36 @@ def val(cls):
 
     @classmethod
     @provide_session
+    def setdefault(cls, key, default, deserialize_json=False, session=None):
+        """
+        Like a Python builtin dict object, setdefault returns the current value
+        for a key, and if it isn't there, stores the default value and returns it.
+
+        :param key: Dict key for this Variable
+        :type key: String
+        :param: default: Default value to set and return if the variable
+        isn't already in the DB
+        :type: default: Mixed
+        :param: deserialize_json: Store this as a JSON encoded value in the DB
+         and un-encode it when retrieving a value
+        :return: Mixed
+        """
+        obj = session.query(cls).filter(cls.key == key).first()
```

Comment:
Would it make sense to reuse the `get` logic directly. Something like:

```python
exists = session.query(literal(True)).where(
    session.query(cls).filter(cls.key == key).exists()
    ).scalar()
if not exists:
    Variable.set(key, default, serialize_json=deserialize_json)
return Variable.get(key, deserialize_json=deserialize_json)
```

(I haven't tested the exist query)